### ISSUE TITLE
Java delivery callback thread issue

### ISSUE THREAD

**Issue by Nook2007** [DoubangoTelecom/ultimateALPR-SDK#212]
Reproduce:
add
```
Log.d(TAG, "thread name=" + Thread.currentThread().getName()+", id = "+Thread.currentThread().getId());
```
to AlprActivity onNewResult method
output:

```
2021-11-26 15:52:04.401 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-54, id = 368754
2021-11-26 15:52:04.471 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-55, id = 368755
2021-11-26 15:52:04.711 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-56, id = 368756
2021-11-26 15:52:04.815 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-57, id = 368757
2021-11-26 15:52:04.901 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-58, id = 368758
2021-11-26 15:52:04.964 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-59, id = 368759
2021-11-26 15:52:05.007 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-60, id = 368760
2021-11-26 15:52:05.038 15605-15642/? D/org.doubango.ultimateAlpr.common.AlprActivity.MyUltAlprSdkParallelDeliveryCallback: thread name=Thread-61, id = 368761
```
New thread created for every callback result

**Comment by DoubangoTelecom** [comment]
The native code to trigger the Java callback looks like this:
```
jvm_->AttachCurrentThread(jenv, NULL);

// <<< JNI call here >>>

jvm_->DetachCurrentThread();
```

Every time `AttachCurrentThread` is called it's up to the JVM  create a new thread or reuse one from a existing pool. So, how new threads are managed it up to the JVM but one clear note is that *there is no leak*.

I have changed the Recognizer program to loop for 10000 times ([Recognizer.txt](https://github.com/DoubangoTelecom/ultimateALPR-SDK/files/7621416/Recognizer.txt)) on Windows and here is what I have noticed when using [ProcessExplorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer):
- There is no memory leak
- The JVM will create multiple threads but the total of the threads for the process is always less than 70 (this number may be different for you but want is important is that it doesn't increase)
- The name of the thread in the callback function is always different but this is meaningless.

In short, the JVM may be creating new threads but what is important is that they are correctly joined and there is no leak. It may be possible to avoid that by using `AttachCurrentThreadAsDaemon` instead of `AttachCurrentThread` but nothing says that it'll improve the performances.
